src/Components/App/App.js

```diff
@@ -9,27 +9,40 @@ class App extends Component {
   constructor() {
     super()
     this.state = {
-      movies: movieData.movies,
+      movies: [],
       movieInfo: null,
+      video: [],
       err: ''
     }
   }
 
+  componentDidMount = () => {
+    const url = `https://rancid-tomatillos.herokuapp.com/api/v2/movies`
+    apiCalls.getMovieData(url)
+    .then(data => this.setState({ movies: data.movies }))
+    .catch(err => this.setState({error: err}))
+  }
+
   getMovieInfo = (id) => {
-    // const url = `https://rancid-tomatillos.herokuapp.com/api/v2${id}`
-    // apiCalls.getSingleMovieData(url)
-    // .then(data => this.setState({ movieInfo: data }))
-    // // .then(data => console.log(data))
-    // .catch((err) => this.setState({error: err}))
+    this.setState( { movies: [] })
+    const url = `https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`
+    apiCalls.getMovieData(url)
+    .then(data => this.setState({ movieInfo: data.movie }))
+    .catch(err => this.setState({error: err}))
+  }
 
-    console.log(id)
-    const matchedMovie = this.state.movies.find(movie => movie.id === id);
-    this.setState({movieInfo: matchedMovie});
-    console.log(this.state.movieInfo)
+  getMovieTrailer = (id) => {
+    const video = `https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}/videos`
+    apiCalls.getMovieData(video)
+    .then(data => this.setState({ video: data.videos.find(video => video.type === "Trailer") }))
+    .catch(err => this.setState({ error: err }))
   }
 
   goToMainView = () => {
     this.setState( { movieInfo: null })
+    const url = `https://rancid-tomatillos.herokuapp.com/api/v2/movies`
```

Comment:
The base of all the URLs is the same. What if you stored that in a global variable in this file that all fetch calls had access to?

Reply:
Duh, great idea. Ty!